src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -246,184 +279,507 @@ private static void VerifyBytes(IntPtr p, int length)
         }
 
         [Fact]
-        public static void ValidateExceptionForHRAPI()
+        public void GetExceptionForHR_ThrowExceptionForHR_ThrowsSameException()
         {
             int errorCode = -2146231029;
             COMException getHRException = Marshal.GetExceptionForHR(errorCode) as COMException;
-            Assert.NotNull(getHRException);
             Assert.Equal(errorCode, getHRException.HResult);
             try
             {
                 Marshal.ThrowExceptionForHR(errorCode);
             }
             catch (COMException e)
-            {             
+            {
                 Assert.Equal(errorCode, e.HResult);
                 Assert.Equal(e.HResult, getHRException.HResult);
             }
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public void GetExceptionForHR_InvalidHR_ReturnsNull(int errorCode)
+        {
+            Assert.Null(Marshal.GetExceptionForHR(errorCode));
+            Assert.Null(Marshal.GetExceptionForHR(errorCode, IntPtr.Zero));
+        }
+
         [Fact]
-        public static void GetHRForException()
+        public void GetHRForException_ValidException_ReturnsValidHResult()
         {
-            Exception e = new Exception();
+            var e = new Exception();
```

Comment:
why change Exception to var?

Reply:
Simple coding conventions - use var when constructing with `new()`